src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2339,6 +2340,38 @@ var projectUniqueNamesForBuildIntToUpdate
                     // Open readme file
                     await OpenReadmeFile(nuGetProject, nuGetProjectContext, token);
                 }
+                catch (SignatureException ex)
+                {
+                    var errors = ex.Results.SelectMany(r => r.GetErrorIssues());
+                    var warnings = ex.Results.SelectMany(r => r.GetWarningIssues());
+                    SignatureException unwrappedException = null;
+
+                    if (errors.Count() == 1)
+                    {
+                        // In case of one error, throw it as the exception
+                        var error = errors.First();
+                        unwrappedException = new SignatureException(error.Code, error.Message, ex.PackageIdentity);
+                    }
+                    else
+                    {
+                        // In case of multiple errors, wrap them in a general NU3000 error
+                        var errorMessage = string.Format(CultureInfo.CurrentCulture,
+                            Strings.SignatureVerificationMultiple,
+                            $"{Environment.NewLine}{string.Join(Environment.NewLine, errors.Select(e => e.FormatWithCode()))}");
+
+                        unwrappedException = new SignatureException(NuGetLogCode.NU3000, errorMessage, ex.PackageIdentity);
+                    }
+
+                    if (warnings.Any())
```

Comment:
Remove this `if` statement.  If `warnings` is empty, the `foreach` below will do nothing.